src/documentBuilder.ts

```diff
@@ -298,7 +299,8 @@ export class DocumentBuilder {
   }
 
   private marshalPermissions() {
-    return {permissions: this.doc.permissions || []};
+    const permissions = this.doc.permissions;
+    return {...(permissions && {permissions})};
```

Comment:
If I understand the issue correctly, an empty array for permissions is always invalid when sending to the API.

If that's the case, and if `marshalPermissions` is meant to represent what's gonna be sent to the API, I think some filtering should be done here (regardless of whether there's also filtering above).

For instance:
```suggestion
    return {...(permissions?.length && {permissions})};
```